realm/realm-library/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2321,4 +2322,97 @@ public void testIsEmpty() {
 
         emptyRealm.close();
     }
+
+    // Test if close can be called from Realm change listener when:
+    // 1. There is an listener on empty Realm Object
+    // 2. There is an listener on non-empty Realm Object
+    // 3. There is an listener on RealmResults
+    // 4. There is no other listeners
+    public void testCloseRealmInChangeListener() {
```

Comment:
The bullets 1-4 seem to indicate it is 4 different tests, which it is not? I would break this unit test into 4 different ones.